aten/src/ATen/native/TensorShape.cpp

```diff
@@ -3302,35 +3302,25 @@ Tensor detach(const Tensor& self) {
     /*allow_tensor_metadata_change=*/false));
 }
 
-Tensor unfold(const Tensor& self, int64_t dimension, int64_t size, int64_t step) {
-  // some special handling to deal with allow dimension == 0 when self.dim() == 0
-  dimension = at::maybe_wrap_dim(dimension, self.dim(), /*wrap_scalar=*/true);
-
-  const auto sizes = self.sizes();
-  const auto strides = self.strides();
-  int64_t max_size = self.dim() == 0 ? 1 : sizes[dimension];
-  TORCH_CHECK(size <= max_size, "maximum size for tensor at dimension ", dimension,
+Tensor unfold(const Tensor& self, int64_t d, int64_t size, int64_t step) {
+  // some special handling to deal with allow d == 0 when self.dim() == 0
+  auto ndim = self.dim();
```

Comment:
Nit: `const auto`

Reply:
I used to use `const` everywhere, but in the end I got bored. To the eyes of the compiler, `auto` already deduces `const`. Now, I think Rust's way is cooler in this regard: const by default! Rust, what a cool language.